Python/ceval.c

```diff
@@ -4323,6 +4403,16 @@ opname ## _miss: \
 
 MISS_WITH_CACHE(LOAD_ATTR)
 MISS_WITH_CACHE(LOAD_GLOBAL)
+MISS_WITH_CACHE(BINARY_SUBSCR)
+
+binary_subscr_error:
```

Comment:
@markshannon I'm not sure I put this in the right place. Maybe it's better to include it at the end of the BINARY_SUBSCR_DICT case and put a `goto error` at the end?  It seem messy to have several error clauses at the end of the switch. What do you think?

Reply:
I prefer the goto. Keeping all the error handling together has advantages for spotting redundancy in the error handling, and makes our intent clear (that error handling is the rare, slow path).

In terms of performance it probably makes no difference as PGO will move it out of line anyway.

Reply:
My concern is not performance but code maintenance. I can leave it here, but still put a (currently redundant) "goto error:" at the end of this block in case more error handling blocks are added in the future.

Reply:
Sure, that sounds good.